tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDITestBase.java

```diff
@@ -110,8 +149,15 @@ public void prepareWorkspace() {
 			cw.open();
 			WebProjectFirstPage fp = new WebProjectFirstPage(cw);
 			fp.setProjectName(PROJECT_NAME);
+			if (CDIVersion.equals("2.0")) {
+				new DefaultCombo(2).setSelection("Dynamic Web Project with CDI 2.0 (Contexts and Dependency Injection)");
+			}
 			fp.setTargetRuntime(sr.getRuntimeName());
-			fp.activateFacet("1.8", "Java");
+			if (isJavaLE8() || CDIVersion.equals("1.0") || CDIVersion.equals("1.2")) {
```

Comment:
This code repeats like 5 times throughout the code. Can you extract it into one place and call it instead?

Reply:
Well, now I see it. Fixed.